aws/aws-toolkit-vscode
# childProcess times out and dies if the debugger does not attach in time

## Types of changes



- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Invoke calls will time out and kill the childProcess if the debugger attachment process takes longer than the value in `aws.samcli.debug.attach.timeout.millis`

## Motivation and Context

Our new output watching approach to debugging leaves the process hanging if the debugger doesn't attach.

## Related Issue(s)

https://github.com/aws/aws-toolkit-vscode/issues/497

## Testing

Manually tested by doing the following:
* Changed the value of `aws.samcli.debug.attach.timeout.millis` to `2` in order to force a timeout and a `childProcess.kill`
* Paused execution on the `waitForDebugPort` call, waited for the `timeoutDuration` to drain to 0, and changed the `debugPort` in order to ensure that the `tcpPortUsed.waitUntilUsed`, and resumed to verify that the `waitUntilUsed` call failed after its initial try

## Checklist





## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
